scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -177,20 +177,38 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
       if (prev.is[Whitespace] || prev.is[Comment]) prev.prev
       else prev
     }
+
+    def nextSafe: Token = scannerTokens.apply(Math.min(token.index + 1, scannerTokens.length - 1))
+
     def next: Token = {
-      val next = scannerTokens.apply(Math.min(token.index + 1, scannerTokens.length - 1))
+      val next = nextSafe
       if (next.is[Whitespace] || next.is[Comment]) next.next
       else next
     }
     def strictNext: Token = {
-      val next = scannerTokens.apply(Math.min(token.index + 1, scannerTokens.length - 1))
+      val next = nextSafe
       if (next.is[Space] || next.is[Tab] || next.is[Comment]) next.strictNext
       else next
     }
   }
 
   /* ------------- PARSER-SPECIFIC TOKENS -------------------------------------------- */
 
+  sealed trait SepRegion {
+    def indent = -1
+    def closeOnNonCase = false
+  }
+  case class RegionIndent(override val indent: Int, override val closeOnNonCase: Boolean)
+      extends SepRegion
+  case object RegionParen extends SepRegion
+  case object RegionBracket extends SepRegion
+  case class RegionBrace(override val indent: Int) extends SepRegion
+  case class RegionEnum(override val indent: Int) extends SepRegion
```

Comment:
Do we need ident here? This should just be a temporary token, that we switch to RegionBrace later and we would calculate ident there.